content/into.md

```diff
@@ -0,0 +1,319 @@
+that is convertible to `IterableOnce[B]`. So the following would work:
+```scala
+scala> val xs = List(1, 2, 3)
+scala> xs.flatMap(x => x.toString * x)
+val res2: List[Char] = List(1, 2, 2, 3, 3, 3)
+```
+Here, the conversion from `String` to `Iterable[Char]` is applied on the results of `flatMap`'s function argument when it is applied to the elements of `xs`.
+
+### Vararg arguments
+
+When applied to a vararg parameter, `into` allows a conversion on each argument value individually. For example, consider a method `concatAll` that concatenates a variable
+number of `IterableOnce[Char]` arguments, and also allows implicit conversions into `IterableOnce[Char]`:
+
+```scala
+def concatAll(xss: into[IterableOnce[Char]]*): List[Char] =
+  xss.foldRight(Nil)(_ ++: _)
+```
+Here, the call
+```scala
+concatAll(List('a'), "bc", Array('d', 'e'))
+```
+would apply two _different_ implicit conversions: the conversion from `String` to `Iterable[Char]` gets applied to the second argument and the conversion from `Array[Char]` to `Iterable[Char]` gets applied to the third argument.
+
+
+### Unwrapping `into`
+
+Since `into[T]` is an opaque type, its run-time representation is just `T`.
+At compile time, the type `into[T]` is a known supertype of the type `T`. So if `t: T`, then
+```scala
+  val x: into[T] = t
+```
+typechecks but
+```scala
+val y: T = x   // error
+```
+is ill-typed. We can recover the underlying type `T` using the `underlying` extension method which is also defined in object `Conversion`:
+```scala
+import Conversion.underlying
+
+val y: T = x.underlying    // ok
+```
+However, the next section shows that unwrapping with `.underlying` is not needed for parameters, which is the most common use case. So explicit unwrapping should be quite rare.
+
+
+
+### Dropping `into` for Parameters in Method Bodies
+
+The typical use cases for `into` wrappers are for parameters. Here, they specify that the
+corresponding arguments can be converted to the formal parameter types. On the other hand, inside a method, a parameter type can be assumed to be of the underlying type since the conversion already took place when the enclosing method was called. This is reflected in the type system which erases `into` wrappers in the local types of parameters
+as they are seen in a method body. Here is an example:
+```scala
+  def ++ (elems: into[IterableOnce[A]]): List[A] =
+    val buf = ListBuffer[A]()
+    for elem <- elems.iterator do // no `.underlying` needed here
+      buf += elems
+    buf.toList
+```
+Inside the `++` method, the `elems` parameter is of type `IterableOnce[A]`, not `into[IterableOne[A]]`. Hence, we can simply write `elems.iterator` to get at the `iterator` method of the `IterableOnce` class.
+
+Specifically, we erase all `into` wrappers in the local types of parameter types that appear in covariant or invariant position. Contravariant `into` wrappers are kept since these typically are on the parameters of function arguments.
```

Comment:
I think the fact that the type of `into` is meaningful from the outside (because of the lower-bound) but needs to be erased inside (because the lower-bound doesn't let you call any method on it) is more confusing than helpful. Either `into` behaves like a normal citizen of the type system and I can understand if something typechecks from looking at the definition of `into` (e.g. if it's `type into[T] = T`), or it's outside the type system (and then it seems like it really is more like a keyword like `=>`).